llvm/lib/Target/AMDGPU/SIInsertWaitcnts.cpp

```diff
@@ -2096,6 +2179,11 @@ bool SIInsertWaitcnts::mayAccessScratchThroughFlat(
   });
 }
 
+bool SIInsertWaitcnts::isVmemAccess(const MachineInstr &MI) const {
+  return (TII->isFLAT(MI) && mayAccessVMEMThroughFlat(MI)) ||
+         (TII->isVMEM(MI) && !AMDGPU::getMUBUFIsBufferInv(MI.getOpcode()));
```

Comment:
`isVMEM` now includes `isFLAT`, so I think this simplifies to:
```suggestion
  return TII->isVMEM(MI) && !AMDGPU::getMUBUFIsBufferInv(MI.getOpcode());
```
which is maybe not what you want because it has lost the `mayAccessVMEMThroughFlat` check. See #137148. Cc @ro-i.